vrrpd/vrrp.c

```diff
@@ -1100,6 +1103,27 @@ static int vrrp_socket(struct vrrp_router *r)
 		goto done;
 	}
 
+	/*
+	 * Bind Tx socket to macvlan device - necessary for VRF support,
+	 * otherwise the kernel will select the vrf device
+	 */
+	if (r->vr->ifp->vrf_id != VRF_DEFAULT) {
+		frr_with_privs (&vrrp_privs) {
```

Comment:
in case vrf backend is netns, should not we prevent to call setsockopt ?

Reply:
The interface bind should still succeed and the program should still behave correctly under netns, we are merely being more paranoid in vrf-lite since kernel source selection will tend to choose the vrf device as the egress device. I prefer not to add special case code for netns.

Reply:
what is the value of r->mvl_ifp->name ?

Reply:
is it possible to add a comment to mention that vrf support does not mean support with vrf netns, please ?

Reply:
Quentin?

Reply:
@pguibert6WIND 
>what is the value of r->mvl_ifp->name ?

`r->mvl_ifp->name` is the macvlan interface name

>is it possible to add a comment to mention that vrf support does not mean support with vrf netns, please ?

Why wouldn't it work under namespace vrf?

Reply:
ok. for this comment, ok.